client/src/lib/offline-storage.ts

```
@@ -1,163 +1,352 @@
+      for (let ncIndex = 0; ncIndex < inspectionData.nonConformities.length; ncIndex++) {
+        const originalNc = inspectionData.nonConformities[ncIndex] as StoreNonConformity;
+        const ncForMainRecord = inspectionDataForMainRecord.nonConformities![ncIndex] as StoreNonConformity;
+
+        if (originalNc.photos && Array.isArray(originalNc.photos)) {
+          for (let photoIndex = 0; photoIndex < originalNc.photos.length; photoIndex++) {
+            const photoRep = originalNc.photos[photoIndex] as PhotoRepresentation;
+            if (photoRep.file) { // If there's a File object, it's a new photo to be saved
+              const photoDbId = await this.savePhotoToDb(localInspectionId, originalNc.id, photoRep.file, photoRep.name);
+
+              // Update the corresponding photo representation in inspectionDataForMainRecord
+              if (ncForMainRecord.photos && ncForMainRecord.photos[photoIndex]) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)